synapse/handlers/room.py

```diff
@@ -1032,7 +1035,11 @@ async def get_event_context(
 
         def filter_evts(events):
             return filter_events_for_client(
-                self.storage, user.to_string(), events, is_peeking=is_peeking
+                self.storage,
+                user.to_string(),
+                events,
+                is_peeking=is_peeking,
+                use_admin_priviledge=use_admin_priviledge,
```

Comment:
Can we not just skip the filtering if `use_admin_priviledge` is set?

Reply:
I decided to go through `filter_events_for_client` because, afaict, it also normalizes events, e.g. by removing redacted events. But now that you mention it, that might not be very useful.

Reply:
I decided to go through `filter_events_for_client` because, afaict, it also normalizes events, e.g. by removing redacted events. But now that you mention it, that might not be very useful.